File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
var areAlmostEqual = function(s1, s2) {
  let [l, r] = [-1, -1];
  
  for (let i = 0; i < s1.length; i++) {
    if (s1[i] === s2[i]) continue;
    if (l === -1) l = i;
    else if (r === -1) r = i;
    else return false;
  }
  
  return s1[l] === s2[r] && s2[l] === s1[r];
};

File: 1790-check-if-one-string-swap-can-make-strings-equal/1790-check-if-one-string-swap-can-make-strings-equal.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        diff = ((x, y) for x, y in zip(s1, s2) if x != y)
        return next(diff, ())[::-1] == next(diff, ()) and not any(diff)
        